### PR TITLE
Filter to sql

### PR DESCRIPTION
**Pull request by Eunovo**
### Description

Implemented a converter for filters to SQL expressions. Tests are include for verification,

##### Checklist

- [x] `npm test` passes
- [x] `npm run build` works
- [x] tests are included

### CODE REVIEW

**Comment by wtrocki** [comment]
Will take look early tommorow

File: packages/datastore/datastore/src/storage/adapters/websql/filterToSQL.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+const defaultOp = (op: string) => (
+    (key: string, value: any) => {
+        if ("string" === typeof value) {
+            value = `'${value}'`;
+        }
+        return `${key} ${op} ${value}`;
+    }
+);
+
+// TODO contains is tricky, verify solution
```
Comment (wtrocki) [inline packages/datastore/datastore/src/storage/adapters/websql/filterToSQL.ts:17]: Was this verified.

File: packages/datastore/datastore/src/storage/adapters/websql/utils.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import invariant from "tiny-invariant";
+import { generateId } from "../..";
+
+// TODO change to constants/enums
+export const prepareStatement = (input: any, type: string = "insert"): [string, any[]] => {
+    if (type === "insert") {
+        input.id = generateId();
```
Comment (wtrocki) [inline packages/datastore/datastore/src/storage/adapters/websql/utils.ts:7]: No the best place to generate id.

**Comment by wtrocki** [comment]
Generating SQL statements seems like very well known and understood problem with handful of the packages and solutions we can use that are mature. Any reason why we went for custom code?

While it is ok to have it written, we need to think about maintenance of that and possible bugs.

**Comment by wtrocki** [comment]
Played with unit tests. Good for merge! Please do follow up improvements as separate PR.

File: packages/datastore/datastore/src/storage/adapters/websql/filterToSQL.ts (hunk 1 above)
Comment (Eunovo) [reply packages/datastore/datastore/src/storage/adapters/websql/filterToSQL.ts:17]: I need to run this to verify it. We still have problem of running sample app

**Comment by Eunovo** [comment]
> Generating SQL statements seems like very well >known and understood problem with handful of the >packages and solutions we can use that are mature. > Any reason why we went for custom code?

TBH, I didn't check for existing packages. I didn't think there would be any tailored to our filters

**Pull request merged**